src/bootstrap/connect.rs

```diff
@@ -54,46 +54,48 @@ async fn tunnel(upgraded: Upgraded, addr: SocketAddr) -> std::io::Result<()> {
 /// This prevents the relay from being used as an arbitrary proxy
 /// to any host on the internet.
 #[instrument]
-fn find_allowable_gateway<B>(req: &Request<B>, gateway_origin: &GatewayUri) -> Option<SocketAddr>
+async fn find_allowable_gateway<B>(
+    req: &Request<B>,
+    gateway_origin: &GatewayUri,
+) -> Option<SocketAddr>
 where
     B: Debug,
 {
     debug!("req: {:?}, gateway_origin: {:?}", req, gateway_origin);
-    if req.uri().authority() != gateway_origin.authority() {
+    if req.uri().authority() != Some(gateway_origin.authority()) {
         debug!("CONNECT request to non-gateway authority: {:?}", req.uri());
         return None;
     }
 
-    uri_to_addr(gateway_origin)
+    gateway_origin.to_socket_addr().await
 }
 
 #[cfg(test)]
 mod test {
-    use http::Uri;
     use once_cell::sync::{Lazy, OnceCell};
     use tracing_subscriber::{self, EnvFilter, FmtSubscriber};
 
     use super::*;
 
     static GATEWAY_ORIGIN: Lazy<GatewayUri> =
-        Lazy::new(|| GatewayUri::new(Uri::from_static("https://0.0.0.0")).unwrap());
+        Lazy::new(|| GatewayUri::from_static("https://0.0.0.0"));
     static INIT: OnceCell<()> = OnceCell::new();
 
-    #[test]
-    fn mismatched_gateways_not_allowed() {
+    #[tokio::test]
+    async fn mismatched_gateways_not_allowed() {
         init_tracing();
         let not_gateway_origin = "https://0.0.0.0:4433";
         let req = hyper::Request::builder().uri(not_gateway_origin).body(()).unwrap();
         let allowable_gateway = find_allowable_gateway(&req, &*GATEWAY_ORIGIN);
-        assert!(allowable_gateway.is_none());
+        assert!(allowable_gateway.await.is_none());
```

Comment:
You mentioned it being strange to assert without context, should a follow up include context?